packages/middleware-stack/src/MiddlewareStack.ts

```diff
@@ -1,381 +1,228 @@
-   *     to specific `step` with priority of `normal`
-   */
-  private getMiddlewareList(): Array<MiddlewareType<Input, Output>> {
-    const middlewareList: Array<MiddlewareType<Input, Output>> = [];
-    const [orphanedRelativeEntries, anchors] = this.normalizeRelativeEntries();
-    let entryList = [...this.absoluteEntries, ...orphanedRelativeEntries];
-    entryList = this.sort(entryList);
-    for (const entry of entryList) {
-      const defaultAnchorValue = { prev: undefined, next: undefined };
-      const { prev, next } = entry.name ? anchors[entry.name] || defaultAnchorValue : defaultAnchorValue;
-      let relativeEntry = prev;
-      //reverse relative entry linked list and add to ordered handler list
-      while (relativeEntry?.prev) {
-        relativeEntry = relativeEntry.prev;
-      }
-      while (relativeEntry) {
-        middlewareList.push(relativeEntry.middleware);
-        relativeEntry = relativeEntry.next;
-      }
-      middlewareList.push(entry.middleware);
-      let orphanedEntry = entry as any;
-      while ((orphanedEntry as any).next) {
-        middlewareList.push((orphanedEntry as any).next.middleware);
-        orphanedEntry = (orphanedEntry as any).next;
-      }
-      relativeEntry = next;
-      while (relativeEntry) {
-        middlewareList.push(relativeEntry.middleware);
-        relativeEntry = relativeEntry.next;
-      }
-    }
-    return middlewareList.reverse();
-  }
-
-  resolve<InputType extends Input, OutputType extends Output>(
-    handler: DeserializeHandler<InputType, OutputType>,
-    context: HandlerExecutionContext
-  ): Handler<InputType, OutputType> {
-    for (const middleware of this.getMiddlewareList()) {
-      handler = middleware(handler as Handler<Input, OutputType>, context) as any;
-    }
-
-    return handler as Handler<InputType, OutputType>;
-  }
+    });
+    const mainChain = sort(normalizedAbsoluteEntries)
+      .map(expandRelativeMiddlewareList)
+      .reduce((wholeList, expendedMiddlewareList) => {
+        // TODO: Replace it with Array.flat();
+        wholeList.push(...expendedMiddlewareList);
+        return wholeList;
+      }, [] as MiddlewareEntry<Input, Output>[]);
+    return mainChain.map((entry) => entry.middleware);
+  };
+
+  const stack = {
+    add: (middleware: MiddlewareType<Input, Output>, options: HandlerOptions & AbsoluteLocation = {}) => {
+      const { name, step = "initialize", tags, priority = "normal" } = options;
+      const entry: AbsoluteMiddlewareEntry<Input, Output> = {
+        name,
+        step,
+        tags,
+        priority,
+        middleware,
+      };
+      if (name) {
+        if (entriesNameSet.has(name)) {
+          throw new Error(`Duplicated middleware name '${name}'`);
+        }
+        entriesNameSet.add(name);
+      }
+      absoluteEntries.push(entry);
+    },
+    addRelativeTo: (middleware: MiddlewareType<Input, Output>, options: HandlerOptions & RelativeLocation) => {
```

Comment:
newline for better readability
```suggestion

    addRelativeTo: (middleware: MiddlewareType<Input, Output>, options: HandlerOptions & RelativeLocation) => {
```

Also on lines 203, 204, 207 and more